data/transactions/logic/eval.go

```diff
@@ -3373,7 +3474,12 @@ func opAppLocalPut(cx *EvalContext) {
 
 	addr, accountIdx, err := cx.accountReference(cx.stack[pprev])
 	if err == nil {
-		err = cx.Ledger.SetLocal(addr, key, sv.toTealValue(), accountIdx)
+		tv := sv.toTealValue()
+		if _, ok := cx.EvalDelta.LocalDeltas[accountIdx]; !ok {
+			cx.EvalDelta.LocalDeltas[accountIdx] = basics.StateDelta{}
+		}
+		cx.EvalDelta.LocalDeltas[accountIdx][key] = tv.ToValueDelta()
+		err = cx.Ledger.SetLocal(addr, key, tv, accountIdx)
```

Comment:
do we need to test matching cx.EvalDelta to Ledger's deltas? Maybe modify `SetLocal` and Co to return  deltas in order to have such logic in a single place?

Reply:
I don't think we can. The reason we're keeping track of them here is that the ledger is accumulating too many updates - it contains updates from inner calls as well.  I suppose we could try to sort it out - reapply them from all the inners and see if it comes out the same. But that feels like a lot of work - interpret all changes twice - to see if they match.